scripts/setup_offline_updates.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+usage() {
+    echo
+    echo "Usage: $0 <repo-type> <tar-file>"
+    echo
+    echo "Creates or updates an XCP-ng offline <repo-type> repository from the <tar-file> archive."
+    echo
+    echo "Arguments:"
+    echo "  repo-type   The repository type, based on its name on online mirrors. Example : 'updates'."
+    echo "  tar-file    The path to the archive containing the repository files."
+}
+
+if [ -z "$2" ]; then
+    echo "Error: argument(s) missing."
+    usage
+    exit 1
+fi
+
+set -eu
+
+REPO_TYPE="$1"
+TAR_FILE="$2"
+REPOS_DIR="/var/local/xcp-ng-repos"
+REPONAME="xcp-ng-offline-$REPO_TYPE"
+
+if [ ! -e "$TAR_FILE" ]; then
+    echo "Error: file $TAR_FILE not found."
+    exit 2
+fi
+
+if ! tar -tf "$TAR_FILE" | grep -q "^${REPO_TYPE}/$"; then
+    echo "Error: directory $REPO_TYPE/ not found in archive $TAR_FILE."
+    exit 3
+fi
+
+mkdir -p "$REPOS_DIR"
+
+if [ -d "$REPOS_DIR/$REPO_TYPE" ]; then
+    echo "- Deleting old repo at $REPOS_DIR/$REPO_TYPE"
+    rm "$REPOS_DIR/$REPO_TYPE" -r
+fi
+
+echo "- Extracting $TAR_FILE to $REPOS_DIR/$REPO_TYPE"
+tar -xf "$TAR_FILE" -C "$REPOS_DIR"
+
+echo "- Writing /etc/yum.repos.d/$REPONAME.repo"
```

Comment:
Those will live alongside `xcp-ng.repo`.  Maybe they should be named `xcp-ng-$REPONAME.repo`, and those files could also be used in the product instead of `xcp-ng.repo`?

Reply:
I'm not sure what you mean. The file name is already `xcp-ng-offline-updates.repo` in the case of `$REPO_TYPE` being `updates`.

Reply:
Oh, I missed that.
The 2nd part of the remark was about whether we really want to keep offline repos *in addition to* networked ones (which require the offline users to disable the networked ones).  Wouldn't it be better to just replace the refs to remote repos with a ref to the offline ones?

Reply:
I haven't put it in the script, but I added the instructions to disable the existing ones at initial pool setup in the related doc I'm writing.